src/common/undo.c

```diff
@@ -282,9 +283,12 @@ static void _undo_do_undo_redo(dt_undo_t *self,
     for(const GList *img = imgs; img; img = g_list_next(img))
```

Comment:
I would avoid 2 list traversal. This should be fully equivalent I think:
```

  for(const GList *img = imgs; img; img = g_list_next(img))
  {
     dt_image_cache_set_change_timestamp(darktable.image_cache, GPOINTER_TO_INT(img->data));

     while(img->next && img->data == img->next->data)
        imgs = g_list_delete_link(imgs, img->next);
  }
```

Reply:
yes :-)